lib/src/rules/no_literal_bool_comparisons.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+import '../analyzer.dart';
+
+const _desc = r"Don't compare booleans to boolean literals.";
+
+const _details = r'''
```

Comment:
We should make it explicit in the docs that the lint doesn't catch cases where `someBool` is nullable.

Reply:
Done